src/login2/mod.rs

```diff
@@ -0,0 +1,142 @@
+mod code;
+mod other_device;
+mod password;
+mod phone_number;
+mod registration;
+
+use glib::subclass::InitializingObject;
+use glib::Properties;
+use gtk::glib;
+use gtk::glib::clone;
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+use gtk::CompositeTemplate;
+use std::cell::Cell;
+
+use crate::components::AnimatedBin;
+use crate::login2::{
+    code::Code, other_device::OtherDevice, password::Password, phone_number::PhoneNumber,
+    registration::Registration,
+};
+use crate::tdlib as model;
+
+mod imp {
+    use super::*;
+
+    #[derive(Debug, Default, CompositeTemplate, Properties)]
+    #[template(string = r#"
+        using Gtk 4.0;
+
+        template Login2 {
+            layout-manager: BoxLayout {
+                orientation: vertical;
+            };
+
+            .AnimatedBin bin { }
```

Comment:
Why not just an AdwLeaflet?

Reply:
It's deprecated and AdwNavigationView isn't well suited to model the tdlib state transitions.